modules/using-an-algorithm/pages/index.adoc

```diff
@@ -0,0 +1,110 @@
+= Using an Algorithm
+:description: Instructions on how to use a GDS algorithm.
+:experimental:
+
+
+This page provides information on how to use a TigerGraph Graph Data Science (GDS) Library algorithm.
+
+All GDS algorithms can be found at https://github.com/tigergraph/gsql-graph-algorithms[`tigergraph/gsql-graph-algorithms`] on GitHub in the `algorithms` folder.
+The algorithms are divided into several categories.
+Each algorithm has its own folder, where you can find the implementation of the algorithm, any user-defined functions required to install the algorithm, or different versions of the algorithm if they exist.
+
+Once you have decided that an algorithm fits your use case, it's time to run the algorithm on your graph data.
+GDS Library algorithms are implemented in GSQL as GSQL queries.
+Therefore, the process to run an algorithm query is the same as the process to run any other query.
+The names of GDS algorithm queries as provided by the library begin with the prefix `tg_` only to distinguish them from a user-written query.
+However, since the library is open-source, you can change the name of the queries to any name of your choosing.
+
+
+== Check schema/data constraints
+Most algorithm queries in the GDS Library are schema-free, meaning that you are able to run the query on any schema.
+However, some algorithms have certain schema or data constraints by nature.
+Make sure to read the documentation for the algorithm to determine the following:
+
+* Does the algorithm require edges to be directed/undirected?
+* Does the algorithm require edges to be weighted/unweighted?
+* Does the algorithm require any vertex type to have an attribute of a certain data type?
+* Does the algorithm expect your data to have been processed in a certain way before it runs?
+
+For example, xref:classification-algorithms:k-nearest-neighbors.adoc[] runs on graphs with either directed or undirected edges, but the edges must have a weight attribute.
+
+Another example is the xref:node-embeddings:fast-random-projection.adoc[Fast Random Projection algorithm], which expects the vertex type to have an attribute of type `LIST<DOUBLE>` if you want to store the embedding results to your graph data.
+
+== Install necessary UDFs
+
+Some algorithms require the use of user-defined functions (UDF).
+The UDF files are included in the same folder as the algorithm itself.
+
+If your query requires a UDF, ensure that the UDF files are added to the GSQL before creating or installing the query.
+If your query does not require a UDF, you can proceed to create the query.
```

Comment:
We are working towards having the algorithms already included in the product. So, rather than saying that "The next step is to create the query", I'd rather say something like
"If the algorithm you wish to use is not yet installed in your TigerGraph instance, you need to install it."
Then explain what are the currently available ways to do that.
The old documentation gave the basic method.
The TigerCLI and the upcoming GraphStudio feature will provide 2 additional ways.

Reply:
I was thinking whenever our algorithm update comes out, we'll update this section accordingly. For now since it isn't out, we describe the steps necessary as they are now.